utils/post.js

```diff
@@ -280,19 +280,21 @@ async function modifyPostLinkPost(domainPageModel, post) {
  * @param {Boolean} [isAnonimous = true]
  * @returns {Object}
  */
-function modifyReactionsPost(post, isAnonimous = true) {
-  if (!isAnonimous) return post;
-
+function modifyReactionsPost(post) {
   const newPost = {...post};
 
-  const itemReducer = (acc, next) => {
+  const itemReducer = (acc, next, isUserDataIncluded = true) => {
     if (next.data.anon_user_info_color_name) {
       next.user = {};
       next.user_id = '';
       next.target_feeds = [];
       next.data.target_feeds = [];
     }
 
+    if (!isUserDataIncluded) {
+      next.user.data = {};
+    }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `itemReducer` function is directly modifying the `next` object which is not a good practice as it can lead to unexpected side effects. Instead, you should create a new object based on `next` and modify that. Also, the `isUserDataIncluded` parameter seems to be used in reverse logic. If `isUserDataIncluded` is true, we would expect user data to be included, but in your code, you're setting `next.user.data = {}` when `isUserDataIncluded` is false. This could lead to confusion for other developers.
```diff
-  const itemReducer = (acc, next, isUserDataIncluded = true) => {
-     if (next.data.anon_user_info_color_name) {
-       next.user = {};
-       next.user_id = '';
-       next.target_feeds = [];
-       next.data.target_feeds = [];
-     }
- 
-    if (!isUserDataIncluded) {
-      next.user.data = {};
-    }
- 
-    acc.push(next);
-    return acc;
-  };
+  const itemReducer = (acc, item, includeUserData = true) => {
+    let newItem = { ...item };
+
+    if (newItem.data.anon_user_info_color_name) {
+      newItem.user = {};
+      newItem.user_id = '';
+      newItem.target_feeds = [];
+      newItem.data.target_feeds = [];
+    }
+
+    if (!includeUserData) {
+      newItem.user.data = {};
+    }
+
+    acc.push(newItem);
+    return acc;
+  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->